src/lib/components/ViewPermissionAddresses.tsx

```diff
@@ -0,0 +1,55 @@
+import { Button } from "@chakra-ui/react";
+import { useState } from "react";
+
+import { useGetAddressType } from "lib/app-provider";
+import type { PermissionAddresses } from "lib/types";
+
+import { ExplorerLink } from "./ExplorerLink";
+import { CustomIcon } from "./icon";
+
+export const ViewPermissionAddresses = ({
+  permissionAddresses,
+}: {
+  permissionAddresses: PermissionAddresses;
+}) => {
+  const [viewAll, setViewAll] = useState(false);
+  const getAddressType = useGetAddressType();
+  const showAddressses =
+    viewAll ||
+    (typeof permissionAddresses === "object" &&
+      permissionAddresses.length === 1);
+
+  return (
+    <>
+      {showAddressses &&
+        permissionAddresses.map((addr) => {
+          return (
+            <ExplorerLink
+              key={addr}
+              type={getAddressType(addr)}
+              value={addr}
+              canCopyWithHover
+            />
+          );
+        })}
+      {permissionAddresses.length > 1 && (
+        <Button
```

Comment:
<img width="256" alt="image" src="https://user-images.githubusercontent.com/14128977/226187356-561bee25-d8f1-427e-9b84-dd96fce762ff.png">

I think the padding is incorrect.